scripts/cmake/git-submodules.cmake

```diff
@@ -5,7 +5,38 @@ set(RIMAGE_CMAKE "${SOF_ROOT_SOURCE_DIRECTORY}/rimage/CMakeLists.txt")
 
 if(GIT_FOUND AND EXISTS "${SOF_ROOT_SOURCE_DIRECTORY}/.git")
 
-	if(NOT EXISTS "${RIMAGE_CMAKE}")
+	if(EXISTS "${RIMAGE_CMAKE}")
+
+		# As incredible as it sounds, some people run neither
+		# "git status" nor "git diff" every few minutes and not
+		# even when their build fails. There has been reports
+		# that they're puzzled when they miss a required
+		# submodule update. This is an attempt to draw their
+		# attention based on the assumption that they pay more
+		# attention to the CMake logs. The warning can be
+		# silenced with I_KNOW_SUBMODULES, see below.
+		execute_process(COMMAND ${GIT_EXECUTABLE} submodule status --recursive
+				WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+				OUTPUT_VARIABLE stdout
+				RESULT_VARIABLE status)
+
+		if(NOT "${status}" EQUAL "0")
+			# stderr was passed-through
+			message(FATAL_ERROR
+"git submodule status --recursive failed, returned: ${status}\n${stdout}")
+		endif()
+
+		if(NOT "$ENV{I_KNOW_SUBMODULES}" AND
```

Comment:
shouldnt the check be at the upper level? if the person makes changes and commits them in the sub repo wont this update the repo on them and undo their changes?

Reply:
> if the person makes changes and commits them in the sub repo wont this update the repo on them and undo their changes?

No, it has never been like this and would not let that happen :-) I've been carrying some local rimage commit hack for ages and I tested this PR with and without it.

In general the build should never ever make any version control change. For convenience the `else` clause below performs the first download but only when `rimage/CMakeLists.txt` is missing.

`I_KNOW_SUBMODULES` is only for people who deviate their submodule(s) on purpose and get annoyed by the warning. I understood this what you asked, was it not?

Reply:
Yep this is what i was clarifying, i misread the command as `update` rather than `status` this makes me much happier